File: src/main/java/com/rs/java/game/area/banks/EdgevilleBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class EdgevilleBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Edgeville Bank";
	}

	public EdgevilleBank() {
		super(
				new Polygon(new WorldTile[] { new WorldTile(3098, 3499, 0), new WorldTile(3098, 3488, 0),
				new WorldTile(3091, 3488, 0), new WorldTile(3091, 3499, 0) }) );
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/EdgevilleBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class EdgevilleBank : Area(
    Polygon(
        arrayOf(
            WorldTile(3098, 3499, 0), WorldTile(3098, 3488, 0),
            WorldTile(3091, 3488, 0), WorldTile(3091, 3499, 0)
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Edgeville Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
